lib/sphinx_rtm/messages.ex

```diff
@@ -18,19 +18,31 @@ defmodule SphinxRtm.Messages do
   def process(message) do
     case Parser.mention_sphinx?(message.text) do
       true ->
-        message
-        |> Map.put(:text, Parser.trim_mention(message.text))
-        |> save_question()
-
         text = Parser.trim_mention(message.text)
 
-        case SlackUtils.search(text, message.channel) do
-          nil ->
-            {:reply, "You asked for \"#{text}\" but I have no answer!"}
+        case Parser.check_content(text) do
+          {:help, _} ->
+            {:reply, help()}
+
+          {:save, content} ->
+            reply =
+              message
+              |> Map.put(:text, content)
+              |> save_question()
+
+            {:reply, reply}
+
+          {:search, content} ->
+            content = if content == "", do: text, else: content
 
-          reply ->
-            {:reply,
-             "You asked for \"#{text}\" but I have no answer, but I found it: \n #{reply}"}
+            case SlackUtils.search(content, message.channel) do
+              nil ->
+                {:reply, "You asked for \"#{content}\" but I have no answer!"}
+
+              reply ->
+                {:reply,
+                 "You asked for \"#{content}\" but I have no answer, but I found it: \n #{reply}"}
```

Comment:
"You asked for \"#{content}\", you might find these helpful: \n #{reply}"

Reply:
Just some text suggestion 😄

Reply:
great